app/components-react/pages/stream-scheduler/StreamScheduler.tsx

```diff
@@ -0,0 +1,242 @@
+  const title = isUpdateMode
+    ? $t('Update Scheduled Stream for %{formattedDate}', { formattedDate })
+    : $t('Schedule Stream for %{formattedDate}', { formattedDate });
+
+  return (
+    <Modal
+      title={title}
+      visible={isModalVisible}
+      onOk={submit}
+      onCancel={closeModal}
+      afterClose={closeModal}
+      destroyOnClose={true}
+      footer={<ModalButtons />}
+      getContainer={`.${css.streamSchedulerPage}`}
+    >
+      <Form form={form}>
+        <Spin spinning={isLoading}>
+          {/* PLATFORM SELECTOR */}
+          {canChangePlatform && (
+            <ListInput
+              label={$t('Platform')}
+              value={selectedPlatform}
+              options={platforms.map(platform => ({
+                value: platform,
+                label: getPlatformDisplayName(platform),
+              }))}
+              onChange={platform => showNewEventModal(platform)}
+            />
+          )}
+
+          {/* TIME PICKER */}
+          <TimeInput name="time" label={$t('Time')} value={time} onChange={setTime} />
+
+          {/* YOUTUBE SETTINGS */}
+          {selectedPlatform === 'youtube' && (
+            <YoutubeEditStreamInfo
+              layoutMode="singlePlatform"
+              isUpdateMode={isUpdateMode}
+              isScheduleMode={true}
+              value={ytSettings}
+              onChange={newSettings => updatePlatform('youtube', newSettings)}
+            />
+          )}
+
+          {/* FACEBOOK SETTINGS */}
+          {selectedPlatform === 'facebook' && (
+            <FacebookEditStreamInfo
+              layoutMode="singlePlatform"
+              isUpdateMode={isUpdateMode}
+              isScheduleMode={true}
+              value={fbSettings}
+              onChange={newSettings => updatePlatform('facebook', newSettings)}
+            />
+          )}
+        </Spin>
+      </Form>
+    </Modal>
+  );
+}
+
+/**
+ * Renders Schedule/Save/Delete buttons
+ */
+function ModalButtons() {
+  const { selectedEvent, remove, submit, isLoading } = useStreamScheduler();
+  const shouldShowSave = !!selectedEvent;
+  const shouldShowSchedule = !selectedEvent;
+
+  // allow removing only those events which the user has not streamed to
+  // removing the event with the finished stream leads to deletion of recorded video too
+  const shouldShowRemove = selectedEvent && selectedEvent.status === 'scheduled';
+
+  /**
+   * confirm and delete
+   */
+  async function onDeleteClick() {
+    if (await confirmAsync($t('Delete the event?'))) remove();
+  }
+
+  return (
+    <Row>
+      <Col flex={'50%'} style={{ textAlign: 'left' }}>
+        {/* DELETE BUTTON */}
+        {shouldShowRemove && (
+          <Button danger onClick={onDeleteClick}>
+            {$t('Delete')}
+          </Button>
+        )}
+      </Col>
+      <Col flex={'50%'}>
+        {/*/!* GO LIVE BUTTON *!/*/}
+        {/*{shouldShowGoLive && <Button type="primary">{$t('Go Live')}</Button>}*/}
```

Comment:
Should this be added back, or intentionally commented out?

Reply:
I think this should be done as a separate PR. It requires some additional changes in our services